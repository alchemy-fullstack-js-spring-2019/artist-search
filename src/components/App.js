import React from 'react';
import ArtistSearch from '../containers/ArtistSearch';
import ArtistById from '../containers/ArtistById';
import ReleaseById from '../containers/ReleaseById';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import RecordingByTitle from '../containers/RecordingByTitle';
import Header from './Header';

export default function App() {
  return (
    <Router>
      <Header />
      <Switch>
        <Route exact path="/" component={ ArtistSearch } />
        <Route exact path="/:artist/:id" component={ ArtistById } />
        <Route exact path="/:artist/release/:id" component={ ReleaseById }/>
        <Route path="/:artist/recording/:title" component={ RecordingByTitle } />
      </Switch>
    </Router>
  );
}
