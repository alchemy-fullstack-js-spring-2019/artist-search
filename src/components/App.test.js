import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/components/App';

describe('App component', () => {
  it('renders the App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
