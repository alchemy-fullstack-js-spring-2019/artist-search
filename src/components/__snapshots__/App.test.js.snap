// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the App 1`] = `
<BrowserRouter>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/:artist/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/:artist/release/:id"
    />
    <Route
      component={[Function]}
      path="/:artist/recording/:title"
    />
  </Switch>
</BrowserRouter>
`;
